### ISSUE TITLE
Unable to connect to mower/cloud

### ISSUE THREAD

**Issue by Trimkap** [geertmeersman/robonect#158]
**Describe the bug**
After udate to 1.64 connection to 'https://[mowername].mower.online' does not work anymore. Have tried to remove hub and unistall integration and restart and then reinstall but it does not help.  Done it many times in different ways. Tried to find some config to remove manually but did not find any.

**To Reproduce**
Add integration
Enter Robonect acccount
Enter Robonect rest account - an url, username and password for automower/mower.online, Hit 'Save'
- Error occurs

**Expected behavior**
The service to be registered and the integration to be able to access the REST API

**Screenshots**
![image](https://github.com/geertmeersman/robonect/assets/119813450/71bc17fa-cdd0-4f07-87b2-a8f885ae0e90)

**Log information**
2024-05-20 13:17:06.405 WARNING (MainThread) [custom_components.robonect] Exception: [Errno -5] Name has no usable address
2024-05-20 13:17:06.415 WARNING (MainThread) [custom_components.robonect] Exception [Errno -5] Name has no usable address
2024-05-20 13:17:07.876 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up robonect.binary_sensor
2024-05-20 13:17:07.879 INFO (MainThread) [homeassistant.components.sensor] Setting up robonect.sensor
2024-05-20 13:17:07.881 INFO (MainThread) [homeassistant.components.button] Setting up robonect.button
2024-05-20 13:17:07.892 INFO (MainThread) [homeassistant.components.device_tracker] Setting up robonect.device_tracker
2024-05-20 13:17:07.892 INFO (MainThread) [homeassistant.components.vacuum] Setting up robonect.vacuum
2024-05-20 13:17:07.893 INFO (MainThread) [homeassistant.components.switch] Setting up robonect.switch
2024-05-20 13:17:07.894 ERROR (MainThread) [homeassistant.components.switch] Error while setting up robonect platform for switch
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 356, in _async_setup_platform
    await asyncio.shield(awaitable)
  File "/config/custom_components/robonect/switch.py", line 108, in async_setup_entry
    for idx, item in enumerate(array):
                     ^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
2024-05-20 13:17:07.901 INFO (MainThread) [homeassistant.components.lawn_mower] Setting up robonect.lawn_mower

<https://www.home-assistant.io/integrations/logger/>

```yaml
logger:
  default: warning
  logs:
    custom_components.robonect: debug
```

**Additional context**

**Comment by geertmeersman** [comment]
Hi, just to be sure, xxx.mower.online is a DNS that you manage locally that points to the IP of the Robonect?

Feel free to join me on discord, which makes it easier to exchange and solve the issue

**Comment by Trimkap** [comment]
Or maybe it is a dyndns service I do not know really 😀

Den 21 maj 2024 12:09, kI 12:09, "Patrik Måneskiöld" ***@***.***> skrev:

**Comment by geertmeersman** [comment]
Ah I didn't even know there was a cloud service :-)
You can pass me the details of course, but the logs should already give some insights

**Comment by Trimkap** [comment]
Here is the function:
https://robonect.de/viewtopic.php?t=3568


Den 21 maj 2024 12:20, kI 12:20, "Patrik Måneskiöld" ***@***.***> skrev:

**Comment by geertmeersman** [comment]
Ok, got it.
Btw, for the host, don't put the https:// in front of it, just the host, so xxxxx.mower.online

**Comment by Trimkap** [comment]
That is true that when it worked I did not use https:// but now I get unknown err and sometimes both errors if that is not there.

![image](https://github.com/geertmeersman/robonect/assets/119813450/c514aa0d-48a4-4580-891a-a2fce0967278)

**Comment by geertmeersman** [comment]
Can you provide me the logs of the debug?

**Comment by Trimkap** [comment]
2024-05-23 12:01:39.443 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
File "/config/custom_components/robonect/config_flow.py", line 266, in test_connection
if not status.get("successful") or status.get("successful") is not True:
^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
resp = await request_handler(request)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
resp = await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 83, in forwarded_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 88, in ban_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/aiohttp_session/__init__.py", line 199, in factory
response = await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 295, in auth_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
response = await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
result = await handler(request, **request.match_info)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 71, in with_admin
return await func(self, request, *args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 222, in post
return await super().post(request, flow_id)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 73, in wrapper
return await method(view, request, data, *args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 122, in post
result = await self._flow_mgr.async_configure(flow_id, data)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 368, in async_configure
result = await self._async_configure(flow_id, user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 414, in _async_configure
result = await self._async_handle_step(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 518, in _async_handle_step
result: _FlowResultT = await getattr(flow, method)(user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/robonect/config_flow.py", line 199, in async_step_connection_rest
test = await self.test_connection(user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/robonect/config_flow.py", line 282, in test_connection
exception = e.message
^^^^^^^^^
AttributeError: 'AttributeError' object has no attribute 'message'
2024-05-23 12:01:49.300 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
File "/config/custom_components/robonect/config_flow.py", line 266, in test_connection
if not status.get("successful") or status.get("successful") is not True:
^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
resp = await request_handler(request)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
resp = await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 83, in forwarded_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 88, in ban_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/aiohttp_session/__init__.py", line 199, in factory
response = await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 295, in auth_middleware
return await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
response = await handler(request)
^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
result = await handler(request, **request.match_info)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 71, in with_admin
return await func(self, request, *args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 222, in post
return await super().post(request, flow_id)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 73, in wrapper
return await method(view, request, data, *args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 122, in post
result = await self._flow_mgr.async_configure(flow_id, data)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 368, in async_configure
result = await self._async_configure(flow_id, user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 414, in _async_configure
result = await self._async_handle_step(
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 518, in _async_handle_step
result: _FlowResultT = await getattr(flow, method)(user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/robonect/config_flow.py", line 199, in async_step_connection_rest
test = await self.test_connection(user_input)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/robonect/config_flow.py", line 282, in test_connection
exception = e.message
^^^^^^^^^
AttributeError: 'AttributeError' object has no attribute 'message'

**Comment by geertmeersman** [comment]
Hi, this is not debug but the error log, you need to enable it on the component as written in the readme :-)

**Comment by Trimkap** [comment]
Or I should say this is what I get when I follow the instructions. :-)

**Comment by geertmeersman** [comment]
Are you able to connect manually to the url?  Is your HA able to connect outside?

**Comment by Trimkap** [comment]
Yes I can ping and curl from the HA ssh terminal. I get the web page from a browser:
![image](https://github.com/geertmeersman/robonect/assets/119813450/60c78ec1-4b55-485c-9ec8-e958fdbf2353)

**Comment by Trimkap** [comment]
I could try.. Is there a link somewhere?

**Comment by geertmeersman** [comment]
On the homepage/readme, but here you go ;-)
https://discord.gg/wZHsA4aGvS

**Comment by geertmeersman** [comment]
Fixed with version v1.6.5
